src/validation/parsePermissions.ts

```diff
@@ -21,211 +21,212 @@ import {
   PermissionIdentityType,
 } from '@coveo/platform-client';
 
-export const processPermissionList = (
-  caseInsensitiveDoc: CaseInsensitiveDocument<PrimitivesValues>,
-  documentBuilder: DocumentBuilder,
-  documentPath: PathLike
-) => {
-  ensurePermissionArray(caseInsensitiveDoc, documentPath);
-  new KnownKeys<Document['permissions']>(
-    'permissions',
-    caseInsensitiveDoc
-  ).whenExists((permissions) => {
-    permissions!.forEach(
-      (permission: Recordable<PermissionSet> | Recordable<PermissionLevel>) => {
-        const caseInsensitivePermission =
-          new CaseInsensitiveDocument<PrimitivesValues>(permission);
-
-        new KnownKeys('permissionsets', caseInsensitivePermission)
-          .whenExists<PermissionLevel>((permissionLevel) =>
-            processPermissionLevel(
-              permissionLevel,
-              documentBuilder,
-              documentPath
+class PermissionParser {
```

Comment:
Refactoring the method to a class to prevent carrying the same parameters to almost every functions